compiler/rustc_trait_selection/src/traits/error_reporting/type_err_ctxt_ext.rs

```diff
@@ -436,22 +435,22 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                         let trait_predicate = self.resolve_vars_if_possible(trait_predicate);
                         let trait_ref = trait_predicate.to_poly_trait_ref();
 
-                        if let Some(_guar) = self.emit_specialized_closure_kind_error(&obligation, trait_ref) {
-                            return;
+                        if let Some(guar) = self.emit_specialized_closure_kind_error(&obligation, trait_ref) {
+                            return guar;
                         }
 
                         // FIXME(effects)
                         let predicate_is_const = false;
 
-                        if self.dcx().has_errors().is_some()
+                        if let Some(guar) = self.dcx().has_errors()
                             && trait_predicate.references_error()
                         {
-                            return;
+                            return guar;
                         }
                         if self.fn_arg_obligation(&obligation) {
                             // Silence redundant errors on binding acccess that are already
                             // reported on the binding definition (#56607).
-                            return;
+                            return self.dcx().span_delayed_bug(obligation.cause.span, "error already reported");
```

Comment:
This is the fishiest place. I can refactor this to make `fn_arg_obligation` contain an `ErrorGuaranteed`

Reply:
ah that was simple, did that

Reply:
Did you? I don't see it.

Reply:
Oh, in the final commit. 👍